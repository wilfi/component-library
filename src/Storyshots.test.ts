import initStoryshots from '@storybook/addon-storyshots';
import { render } from '@testing-library/react';

/**
 * This is a turnaround needed for the module whatwg-url error
 * `TextEncoder is not defined Reference`
 * Which the following solution can be found here:
 * https://github.com/jsdom/whatwg-url/issues/209
 *
 * But since we're assigning TextDecoder a different type
 * from global.TextDecoder, we add a @ts-ignore into
 * that line
 */
import { TextEncoder, TextDecoder } from 'util';
global.TextEncoder = TextEncoder;
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
global.TextDecoder = TextDecoder;

const reactTestingLibrarySerializer: jest.SnapshotSerializerPlugin = {
  print: (val: unknown, serialize: (i: unknown) => string) =>
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    serialize(val?.container?.firstChild),
  test: (val) => val && val.hasOwnProperty('container'),
};

initStoryshots({
  renderer: render,
  storyKindRegex: /^Widgets\//,
  snapshotSerializers: [reactTestingLibrarySerializer],
});
