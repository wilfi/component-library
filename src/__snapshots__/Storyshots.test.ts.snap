// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Widgets/Emoji Cards/Components Cat 1`] = `
<div
  role="main"
  style="display: flex; flex-direction: column; align-items: center; margin: 5px;"
>
  <div
    style="position: relative; width: 100%; z-index: 0;"
  >
    <div
      style="background: white; border-radius: 0.5rem; box-shadow: 0.125rem 0.125rem 0.5rem rgba(0, 0, 0, 0.1); display: flex; font-family: helvetica, sans-serif; overflow: hidden; max-width: 26rem; height: 13rem;"
    >
      <div
        style="background: rgba(0, 0, 0, 0.1); display: flex; flex-direction: column; font-size: 8rem; justify-content: center; line-height: 100%; overflow: hidden; position: relative; padding: 1rem; background-color: rgba(0, 255, 255, 0.3);"
      >
        <span
          style="text-shadow: 0 0 1rem rgba(0, 0, 0, 0.5); z-index: 1;"
        >
          üê±
        </span>
        <span
          style="left: -1rem; right: -2rem; top: 0px; bottom: 0px; font-size: 5rem; line-height: 3rem; position: absolute; word-break: break-all;"
        >
          catcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcatcat
        </span>
      </div>
      <div
        style="background: rgba(0, 0, 0, 0.01); color: rgba(0, 0, 0, 0.6); display: flex; flex: 1 1 0%; flex-direction: column; padding: 1rem;"
      >
        <span
          style="font-weight: bold; font-size: 2rem;"
        >
          cat
        </span>
        <span
          style="flex: 1 1 0%; font-size: 1.25rem; padding: 1rem 0px;"
        >
          ‚Äú
          Cat face
          ‚Äù
        </span>
        <span
          style="color: rgba(0, 0, 0, 0.4); font-size: 0.85rem; text-transform: uppercase; text-align: right;"
        >
          Animals & Nature
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Widgets/Emoji Cards/Components Invalid 1`] = `
<div
  role="main"
  style="display: flex; flex-direction: column; align-items: center; margin: 5px;"
>
  <div
    style="position: relative; width: 100%; z-index: 0;"
  >
    <div
      style="background: white; border-radius: 0.5rem; box-shadow: 0.125rem 0.125rem 0.5rem rgba(0, 0, 0, 0.1); display: flex; font-family: helvetica, sans-serif; overflow: hidden; max-width: 26rem; height: 13rem;"
    >
      <div
        style="background: rgba(0, 0, 0, 0.1); display: flex; flex-direction: column; font-size: 8rem; justify-content: center; line-height: 100%; overflow: hidden; position: relative; padding: 1rem; background-color: rgba(255, 255, 0, 0.3);"
      >
        <span
          style="text-shadow: 0 0 1rem rgba(0, 0, 0, 0.5); z-index: 1;"
        >
          üö´
        </span>
        <span
          style="left: -1rem; right: -2rem; top: 0px; bottom: 0px; font-size: 5rem; line-height: 3rem; position: absolute; word-break: break-all;"
        >
          invalidinvalidinvalidinvalidinvalidinvalidinvalidinvalidinvalidinvalidinvalidinvalidinvalidinvalidinvalidinvalidinvalidinvalidinvalidinvalidinvalidinvalidinvalidinvalidinvalidinvalidinvalidinvalidinvalid
        </span>
      </div>
      <div
        style="background: rgba(0, 0, 0, 0.01); color: rgba(0, 0, 0, 0.6); display: flex; flex: 1 1 0%; flex-direction: column; padding: 1rem;"
      >
        <span
          style="font-weight: bold; font-size: 2rem;"
        >
          invalid
        </span>
        <span
          style="flex: 1 1 0%; font-size: 1.25rem; padding: 1rem 0px;"
        >
          ‚Äú
          Empty
          ‚Äù
        </span>
        <span
          style="color: rgba(0, 0, 0, 0.4); font-size: 0.85rem; text-transform: uppercase; text-align: right;"
        >
          Symbols
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Widgets/Emoji Cards/Components Plus One 1`] = `
<div
  role="main"
  style="display: flex; flex-direction: column; align-items: center; margin: 5px;"
>
  <div
    style="position: relative; width: 100%; z-index: 0;"
  >
    <div
      style="background: white; border-radius: 0.5rem; box-shadow: 0.125rem 0.125rem 0.5rem rgba(0, 0, 0, 0.1); display: flex; font-family: helvetica, sans-serif; overflow: hidden; max-width: 26rem; height: 13rem;"
    >
      <div
        style="background: rgba(0, 0, 0, 0.1); display: flex; flex-direction: column; font-size: 8rem; justify-content: center; line-height: 100%; overflow: hidden; position: relative; padding: 1rem; background-color: rgba(255, 255, 0, 0.3);"
      >
        <span
          style="text-shadow: 0 0 1rem rgba(0, 0, 0, 0.5); z-index: 1;"
        >
          üëç
        </span>
        <span
          style="left: -1rem; right: -2rem; top: 0px; bottom: 0px; font-size: 5rem; line-height: 3rem; position: absolute; word-break: break-all;"
        >
          +1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1+1
        </span>
      </div>
      <div
        style="background: rgba(0, 0, 0, 0.01); color: rgba(0, 0, 0, 0.6); display: flex; flex: 1 1 0%; flex-direction: column; padding: 1rem;"
      >
        <span
          style="font-weight: bold; font-size: 2rem;"
        >
          +1
        </span>
        <span
          style="flex: 1 1 0%; font-size: 1.25rem; padding: 1rem 0px;"
        >
          ‚Äú
          Thumbs up sign
          ‚Äù
        </span>
        <span
          style="color: rgba(0, 0, 0, 0.4); font-size: 0.85rem; text-transform: uppercase; text-align: right;"
        >
          People & Body
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Widgets/Emoji Cards/Components Spanish 1`] = `
<div
  role="main"
  style="display: flex; flex-direction: column; align-items: center; margin: 5px;"
>
  <div
    style="position: relative; width: 100%; z-index: 0;"
  >
    <div
      style="background: white; border-radius: 0.5rem; box-shadow: 0.125rem 0.125rem 0.5rem rgba(0, 0, 0, 0.1); display: flex; font-family: helvetica, sans-serif; overflow: hidden; max-width: 26rem; height: 13rem;"
    >
      <div
        style="background: rgba(0, 0, 0, 0.1); display: flex; flex-direction: column; font-size: 8rem; justify-content: center; line-height: 100%; overflow: hidden; position: relative; padding: 1rem; background-color: rgba(0, 255, 0, 0.3);"
      >
        <span
          style="text-shadow: 0 0 1rem rgba(0, 0, 0, 0.5); z-index: 1;"
        >
          üá™üá∏
        </span>
        <span
          style="left: -1rem; right: -2rem; top: 0px; bottom: 0px; font-size: 5rem; line-height: 3rem; position: absolute; word-break: break-all;"
        >
          flag-esflag-esflag-esflag-esflag-esflag-esflag-esflag-esflag-esflag-esflag-esflag-esflag-esflag-esflag-esflag-esflag-esflag-esflag-esflag-esflag-esflag-esflag-esflag-esflag-esflag-esflag-esflag-esflag-es
        </span>
      </div>
      <div
        style="background: rgba(0, 0, 0, 0.01); color: rgba(0, 0, 0, 0.6); display: flex; flex: 1 1 0%; flex-direction: column; padding: 1rem;"
      >
        <span
          style="font-weight: bold; font-size: 2rem;"
        >
          flag-es
        </span>
        <span
          style="flex: 1 1 0%; font-size: 1.25rem; padding: 1rem 0px;"
        >
          ‚Äú
          Bandera de Espa√±a
          ‚Äù
        </span>
        <span
          style="color: rgba(0, 0, 0, 0.4); font-size: 0.85rem; text-transform: uppercase; text-align: right;"
        >
          Banderas
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Widgets/My Widget/Components/MyWidget Default 1`] = `
<div
  role="main"
  style="display: flex; flex-direction: column; align-items: center; margin: 5px;"
>
  <div
    style="position: relative; width: 100%; z-index: 0;"
  >
    <div
      style="width: 15em; background-color: white; border-radius: 6px; box-shadow: 1px 1px 5px rgba(0, 0, 0, 0.3); padding: 2em;"
    >
      <h5
        style="font-size: 1.4em; font-weight: bold; text-decoration: underline; margin-bottom: 0.7em;"
      >
        Greeting
      </h5>
      <div>
        Hello
        , 
        <em>
          Alice
        </em>
        .
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Widgets/My Widget/Components/MyWidget Hi 1`] = `
<div
  role="main"
  style="display: flex; flex-direction: column; align-items: center; margin: 5px;"
>
  <div
    style="position: relative; width: 100%; z-index: 0;"
  >
    <div
      style="width: 15em; background-color: white; border-radius: 6px; box-shadow: 1px 1px 5px rgba(0, 0, 0, 0.3); padding: 2em;"
    >
      <h5
        style="font-size: 1.4em; font-weight: bold; text-decoration: underline; margin-bottom: 0.7em;"
      >
        Greeting
      </h5>
      <div>
        Hi
        , 
        <em>
          Alice
        </em>
        .
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Widgets/My Widget/Components/MyWidget Howdy 1`] = `
<div
  role="main"
  style="display: flex; flex-direction: column; align-items: center; margin: 5px;"
>
  <div
    style="position: relative; width: 100%; z-index: 0;"
  >
    <div
      style="width: 15em; background-color: white; border-radius: 6px; box-shadow: 1px 1px 5px rgba(0, 0, 0, 0.3); padding: 2em;"
    >
      <h5
        style="font-size: 1.4em; font-weight: bold; text-decoration: underline; margin-bottom: 0.7em;"
      >
        Greeting
      </h5>
      <div>
        Howdy
        , 
        <em>
          Alice
        </em>
        .
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Widgets/My Widget/Components/MyWidget Spanish 1`] = `
<div
  role="main"
  style="display: flex; flex-direction: column; align-items: center; margin: 5px;"
>
  <div
    style="position: relative; width: 100%; z-index: 0;"
  >
    <div
      style="width: 15em; background-color: white; border-radius: 6px; box-shadow: 1px 1px 5px rgba(0, 0, 0, 0.3); padding: 2em;"
    >
      <h5
        style="font-size: 1.4em; font-weight: bold; text-decoration: underline; margin-bottom: 0.7em;"
      >
        Saludo
      </h5>
      <div>
        Hi
        , 
        <em>
          Al√≠cia
        </em>
        .
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Widgets/My Widget/Components/MyWidget With Excitement 1`] = `
<div
  role="main"
  style="display: flex; flex-direction: column; align-items: center; margin: 5px;"
>
  <div
    style="position: relative; width: 100%; z-index: 0;"
  >
    <div
      style="width: 15em; background-color: white; border-radius: 6px; box-shadow: 1px 1px 5px rgba(0, 0, 0, 0.3); padding: 2em;"
    >
      <h5
        style="font-size: 1.4em; font-weight: bold; text-decoration: underline; margin-bottom: 0.7em;"
      >
        Greeting
      </h5>
      <div>
        Howdy
        , 
        <em>
          Alice
        </em>
        !
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Widgets/My Widget/Components/MyWidget With Excitement Emoji 1`] = `
<div
  role="main"
  style="display: flex; flex-direction: column; align-items: center; margin: 5px;"
>
  <div
    style="position: relative; width: 100%; z-index: 0;"
  >
    <div
      style="width: 15em; background-color: white; border-radius: 6px; box-shadow: 1px 1px 5px rgba(0, 0, 0, 0.3); padding: 2em;"
    >
      <h5
        style="font-size: 1.4em; font-weight: bold; text-decoration: underline; margin-bottom: 0.7em;"
      >
        Greeting
      </h5>
      <div>
        Hi
        , 
        <em>
          üéâ
        </em>
        !
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Widgets/My Widget/Components/MyWidget With Excitement Hi 1`] = `
<div
  role="main"
  style="display: flex; flex-direction: column; align-items: center; margin: 5px;"
>
  <div
    style="position: relative; width: 100%; z-index: 0;"
  >
    <div
      style="width: 15em; background-color: white; border-radius: 6px; box-shadow: 1px 1px 5px rgba(0, 0, 0, 0.3); padding: 2em;"
    >
      <h5
        style="font-size: 1.4em; font-weight: bold; text-decoration: underline; margin-bottom: 0.7em;"
      >
        Greeting
      </h5>
      <div>
        Hi
        , 
        <em>
          Bob
        </em>
        !
      </div>
    </div>
  </div>
</div>
`;
